SecurityPkg/Library/Tcg2InitEventLib/Tcg2InitEventLib.c

```diff
@@ -0,0 +1,143 @@
+/** @file
+  Initializes TCG event data structures
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+#include <Library/Tpm2CommandLib.h>
+#include <Library/DebugLib.h>
+#include <Library/BaseLib.h>
+#include <Library/UefiLib.h>
+#include <Library/UefiLib/UefiLibInternal.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/Tcg2InitEventLib.h>
+#include <IndustryStandard/UefiTcgPlatform.h>
+
+/**
+  This function initializes the TCG_PCR_EVENT2_HDR for EV_NO_ACTION events. The behavior is defined
+  by TCG PC Client PC Client PFP SPec. Secion 9.3.4 EV_NO_ACTION Event Types.
+
+  Note: The EFI Specification ID event is not supported.
+
+  It is considered an error to call this function with a NULL pointer, or an ActivePcrBanks value of 0.
+
+  @param[in, out]   Event           Event Header of an EV_NO_ACTION type event
+  @param[in]        EventSize       Size of the Event argument

```

Comment:
this implementation never checks EventSize to make sure Event is large enough of a memory allocation.